### ISSUE TITLE
SiriProxy stopped working, banned?

### ISSUE THREAD

**Issue by RonaldKonijnenburg** [plamoni/SiriProxy#323]
Hi guys,

I am using Siri Proxy on a cloud server hosted by Cloudshare.  Also a VPN server is installed so that I can use siri proxy while on the road.

Siri Proxy has been running fine for  a month, but for the last couple of days, but suddenly stopped working (never a good sign).  I have tried everything to fix the error . Got a new external IP address, reinstalled siri proxy, refreshed certificates, rebundled, restarted DNSmasq you name it. It simply does not work anymore.

Log file show:
# 

[Info - iPhone] Received Object: FinishSpeech (group: com.apple.ace.speech, ref_id: 158BCD1E-C3E1-4C4B-91FF-9BD48AF08118, ace_id: 891E4177-0716-470F-9040-7B75D513287C)
{"class"=>"FinishSpeech",
 "refId"=>"158BCD1E-C3E1-4C4B-91FF-9BD48AF08118",
 "group"=>"com.apple.ace.speech",
 "aceId"=>"891E4177-0716-470F-9040-7B75D513287C",
 "properties"=>{"packetCount"=>16}}
[Info - Forwarding object to Guzzoni] FinishSpeech
[Debug - iPhone] Forwarding 96 bytes of data to Guzzoni
[Ping - iPhone](4)
[Debug - iPhone] Forwarding 10 bytes of data to Guzzoni
========UNZIPPED DATA (from iPhone =========
#03 00 00 00 05

[Ping - iPhone](5)
[Debug - iPhone] Forwarding 9 bytes of data to Guzzoni
========UNZIPPED DATA (from iPhone =========
#03 00 00 00 06

[Ping - iPhone](6)
[Debug - iPhone] Forwarding 9 bytes of data to Guzzoni
========UNZIPPED DATA (from iPhone =========
03 00 00 00 07
=============================================s=====
[Ping - iPhone](7)
[Debug - iPhone] Forwarding 9 bytes of data to Guzzoni
========UNZIPPED DATA (from iPhone =========
03 00 00 00 08

etc, etc. It finally times out at ping 16. I have a ping, but not a pong. I believe Apple servers are not responding.

Do you guys have any idea where to look? I am starting to think my server is blocked by Apple (how is beyond me, guess they must have blocked a range of IPs if that really is the case). 

Can you guys give me an idea where to look further?

Thanks,

sjako.

**Comment by plamoni** [comment]
I had a little hiccup with mine. I re-routed so that I wasn't using
SiriProxy (for you that probably means disconnecting from the VPN) and used
Siri a couple times and then things started working again with SiriProxy.

I should note that SiriProxy is not designed to assist in using Siri on a
non-iPhone 4S device. Some folks have modified it for that use, but that's
not it's basic use. Right now, it appears people who are trying to use Siri
on non-4S devices are having trouble. If you're trying to use it on a
non-4S device then I can't really help you. But if you're using a 4S, then
it should be working as normal. Try using Siri without SiriProxy a couple
times and see if that corrects the problem.

-Pete

**Comment by RonaldKonijnenburg** [comment]
Hi Pete,

Strictly iPhone 4S. No tampering with iPhone 4. I therefore also would not understand Apple blockin the IP. On the other hand how would they know that I am using it in a legal manner....

Will try using siri without the proxy for some time and post the results.

**Comment by plamoni** [comment]
Sounds good.

I think something about using SiriProxy prevents Apple's remote update
process from working properly, which is why things stopped working for you
(and me for a little bit). So it's not that there's anything wrong with
SiriProxy. Only that Apple can't update the app remotely for some reason
when you're using it. It's possible they're using a special kind of
response that's not in our known collection (packets starting with "00" and
"01" are conspicuously absent from the packets we see regularly and I don't
know how to handle them if they are used by Apple). Anyway, hopefully your
phone will figure itself out by bypassing SiriProxy for a little bit and
then you can go back to using it.

Best of luck. Let me know how it goes!

-Pete

**Comment by RonaldKonijnenburg** [comment]
Makes sense. I sure hope it will work out by itself.

I did use a different 4S yesterday on the server. That also did not work. So weird..., 

 Thanks for your helping and building this awesome software!

**Comment by RonaldKonijnenburg** [comment]
Pete,

Just to let you know. I was blocked by Apple and was not using illegal (non 4S) devices. Go figure.

I managed to spoof my IP and DNS by using proxychains and Tor. After installing those, she runs beautifully again. Used this tutorial:

http://forum.intern0t.org/security-tutorials-guides/3787-proxychains-tor.html

**Comment by plamoni** [comment]
Interesting.

If Apple is blocking your IP address it's likely only temporary (they
wouldn't want to risk blocking the next person to get your IP). If they
wanted to block your device, they could blacklist your UDID and that would
stop you from using Siri altogether -- at least not without spoofing your
authentication data like people do with non-4S devices.

The only reason I can see Apple temp-blocking your IP is if it was causing
trouble. For instance if you continually sent malformed packets or did
something else disruptive to their servers. I did this a lot when creating
SiriProxy and they never bothered to block my IP. This would be the first
instance of them doing any sort of blocking that I've heard of, actually.

I think it's more likely that something about your setup is broken and
that's causing SiriProxy to not work for you. I recommend doing some more
trouble shooting of your setup. If it still looks like you're blocked, just
wait a couple days and try again.

-Pete

On Fri, Feb 17, 2012 at 3:08 PM, sjako <
reply@reply.github.com

**Comment by gotnull** [comment]
Seems I can't do anything with SiriProxy at the moment as well. I see speech packets being processed however it just sits at FinishSpeech now and doesn't go anywhere from this point onwards.

I have an iPhone 4S and I've never used anything other than this device. I don't use a VPN or anything fancy like that either. This was working a couple of weeks ago and I hadn't touched anything on the configuration side since then.

Very strange.

**Comment by RonaldKonijnenburg** [comment]
The plot thickens ;-)

**Comment by gotnull** [comment]
Sure does. I'll give it another try when I get home. Something is definitely not right though. I've never had a problem before.

**Comment by SnecxXx** [comment]
Hello

Yesterday i installed the proxy. I got it working. I tested some scripts and went to bed. I woke up at around 1 o'clock AM.
Just 4 fun i tried siri. Not working anymore. The same issue like @rakusu. 

Today at 2 PM  still no luck. The last log entry is 'Finished Speech' but no response from the apple server.

Now at 2:20 PM Siri proxy is working again. My IP is the same like yesterday. I use the proxy with a 4s only.

Any clue whats causing the problem? It seems i was blocked by apple.

**Comment by plamoni** [comment]
Guess what? Mine stopped working today, too. Works fine on 3G. No response
from Guzzoni through SiriProxy.

Something strange is happening.

-Pete

On Sat, Feb 18, 2012 at 7:27 AM, Snecx <
reply@reply.github.com

**Comment by plamoni** [comment]
Tested through WiFi without SiriProxy turned on. Still no dice. I think
they're temp-banning IP addresses that use SiriProxy for some reason. Not
sure how they know (there are a couple ways they could tell that I can
think of). But you're right, the plot is thickening.

-Pete

**Comment by RonaldKonijnenburg** [comment]
Wow, I wonder what is behind all this. Would be a shame if we loose this great functionality.

**Comment by notmyname4** [comment]
Same here. I was in the middle of testing a plugin and it started hanging on FinishSpeech.
Here's the twist.
My rig is a Dell Laptop (Ubuntu) which I was testing at work. Took it home that day, just a few hours after the issue started, and it still would hang. Obviously the external ip is different between the two, so don't think it's the ip they're after.
I had to remove the ca from my 4s for Siri to work again, even while on 3G. Maybe this is what apple is focused on?

In a hasty decision, I formated and started from the beginning. Now I'm running into the "/usr/bin/env: siriproxy: No such file or directory" issue. I now have two laptops dedicated to siriproxy and they both are giving me trouble. Arg!

**Comment by TobyTech** [comment]
No issues at all so far 0_o. I just jinxed myself.

**Comment by SnecxXx** [comment]
if you wait some hours the proxy works again for 1or 2 commands.
after that it stops working again.
any clue why it happens to a very mall group of proxys?
i don't cause much traffic to the apple servers. i guess the comercial servers are causing a lot more traffic and i read nothing that the servers are down.
this is bullsh..

**Comment by notmyname4** [comment]
I've made it past the "/usr/bin/env: siriproxy: No such file or directory" issue. Thanks to https://github.com/plamoni/SiriProxy/issues/318

No issues to report regarding blocking. I've been talking to Siri quite heavily and I'm not hanging on FinishSpeech anymore (for now?). So back to work on my plugin. Let's just hope that Apple is not responsible for the recent weirdness, and that they see SiriProxy as helping to sell more iDevices. Siri on iPad 3 will make me forget about the large amount of money I spent on the iPad2 and fork out another large amount of money yet again, if I can still use SiriProxy.

**Comment by SnecxXx** [comment]
Still have the blocking issue. If i delete the file com.apple.assistant.plist as suggested in some forums i'm able to talk to siri again. Every 2 or 3 times it ends with a loop of ping. 
If you try again to talk to siri it works. 
Any help i can provide to fix this issue? I love this project and i want to integrate siri proxy to my home automation system.

**Comment by RonaldKonijnenburg** [comment]
Ignore this post.

**Comment by RonaldKonijnenburg** [comment]
I am also getting blocked now while using proxychains. Seems it is not strictly banning on IP (anymore).

**Comment by notmyname4** [comment]
Question
Is everyone using the default SiriProxyCA info? 
Could be they are hunting down profiles that match the default info or CA name?

**Comment by RonaldKonijnenburg** [comment]
Good suggestion. I am using the default one (like everybody does I guess).

**Comment by Ponyboy47** [comment]
My iphone 4s is not using the default SiriProxyCA name or info but my proxy is blocked too. I can use siri just fine without the proxy but it always hangs on FinishSpeech or gives the "I'm sorry but I'm unavailable right now.." response

**Comment by Ponyboy47** [comment]
but the certificate I use is new. I was having issues with my proxy from time to time and since I'd been tampering with the code I figured I might have messed something up so I erased my proxy and just started over. I use the westbaer fork of siriproxy using dynamic dns...this allows me to use my proxy without having to have set up a vpn server. And so my ca.crt name is different as is some of the info in my certificate

**Comment by Ponyboy47** [comment]
Sorry should've specified that it allows me to use my proxy over 3g without having to use a vpn server

**Comment by gotnull** [comment]
I'd say these reports are pretty much a confirmation that some sort of blocking is happening from Apple specifically targeted to those of us that use SiriProxy. Siri is working just fine for me - however SiriProxy still remains frozen on FinishSpeech. I'm using the default CA certificate generated by the installation and I'm running Ubuntu 11.10 not that it would make a difference what OS you're using.

This, sucks.

**Comment by Ponyboy47** [comment]
Until SiriProxy is working again I'd look into AssistantExtensions...for those who have jailbroken their 4S anyways..

**Comment by notmyname4** [comment]
Update.

Started hanging on FinishSpeech after I used Siri on 3G (got directions to a pizza place). When I returned home the issue started. I changed my dns back to the router and used Siri to check the weather, switched dns back to SiriProxy and all is well. Didn't touch the ca, just the dns.

**Comment by gotnull** [comment]
That's a good sign, it could be that Apple is making changes to the Guzzoni server and it's affecting SiriProxy in particular. Might wait another few days and see if it's still happening for me. I'll refrain from using Siri at all until then whether it's 3G or Wifi.

**Comment by RonaldKonijnenburg** [comment]
Is this a DNS issue? Would that be the reason that the three little piggies proxy does not have a problem as it is using Spire?

**Comment by gotnull** [comment]
Don't know much about Spire to be honest. I've got an iPhone 4S.

**Comment by Ponyboy47** [comment]
Spire is the jailbreak app that installs siri on non 4S devices and automatically forwards siri requests to a server of your choice. I use it on my 4S to forward my siri to a dyndns server i run at home that forwards all siri requests to my proxy server. Which is a lot easier than setting up a vpn server in order to have your siriproxy away from home

**Comment by cnnn** [comment]
Probably something failed to be sent back to apple server because DNS can't resolve some host name.

**Comment by RonaldKonijnenburg** [comment]
@cnnn any idea where I need to check this? resolv.conf?

**Comment by cnnn** [comment]
@sjako check server log, or trying putting the host in assistant.plist instead of changing dns.

**Comment by RonaldKonijnenburg** [comment]
Put in the DNS in plist, endless restart occurred while on VPN.

**Comment by Lunat1k** [comment]
I notice that the 4S sends an HTTP header of X-ACE-HOST over. however I didn't check to see if siriproxy is doing the same. I have a feeling Apple is looking for that HTTP header now.

**Comment by RonaldKonijnenburg** [comment]
Hmmm, think the three little pigs does not do that. At least not for non 4S devices. Anybody know how to not send them?

**Comment by gotnull** [comment]
Three little pigs? What do you mean?

**Comment by RonaldKonijnenburg** [comment]
Three little pigs, an alternative proxy:

https://github.com/jimmykane/The-Three-Little-Pigs-Siri-Proxy

Readme:

A little info upon public/semipublic servers

So you got a server and you are afraid if apple will ban your device? Lets get things straight then

```
The Three Little Pigs DO NOT send any XACE-HOST, assistantid,speechid to Apple servers. They also get right of the host header.

Apple doesnt ban by UUID (as far as now!)

Apple bans by IP! So use proxychains/Tor/tsocks/torsocks/torify etc to fake the ip and DNS

Also keep in mind that there is a limit on how many assistants you can create per key per day. I am not sure but 30 is an average. Please correct me if I am mistaken. There will be a new release soon where there will be a limit on that and hopefully this issue will be resolved.
```

**Comment by Lunat1k** [comment]
FYI I think the X-ACE-HOST is the factor in this because from my wifi connection my outgoing ip address is exactly the same between the phone and my proxy. From the phone directly to apple it works, but from phone to proxy to apple it does not. If my IP was banned then I wouldn't even be able to use siri directly pointing to apple.

**Comment by cnnn** [comment]
I also find that whenever a 4S connects with "CreateAssistant" sent first, it will ALWAYS receive responds from Apple, in which case it does not send X-Ace-Host because it doesn't have one.
But after omitting X-Ace-Host, the server only receives "SetActivationToken".

**Comment by amilham** [comment]
My SiriProxy has stopped working also (this is on a real, non-jailbroken, iPhone 4S). Just stops at "FinishSpeech".

**Comment by ghost** [comment]
I've been having the same problem over the past week or so. When I experience it, I switch off my wifi and use siri over 3G talking directly to apple, and ask for the weather a couple of times. After 2 or 3 attempts it starts to work again. If I then reconnect to my wifi and use SiriProxy, it works fine for a few days.

Not too bad of a workaround, but I hope someone is able to find a fix for this soon.

**Comment by RonaldKonijnenburg** [comment]
Can someone confirm if @mtrayn01 workaround works? Cannot try to
do it myself as I would need to re-install my server.

**Comment by RonaldKonijnenburg** [comment]
Crazy: siri without siri proxy:

VPN to my server on WIFi --> Siri fails
Three times what's the weather without VPN --> Siri works
VPN to my server on WIFi again --> Siri fails
VPN to my server on 3G --> Siri fails
Three times what the weather without VPN (wifi/3g does not matter) --> Siri works
VPN to my server on 3G --> Siri works
VPN to my server on wifi --> siri fails

Conclusion: they won, I'm lost.

**Comment by Lunat1k** [comment]
Hmm I just confirmed mtrayn01's trick sort of works. I was able to get a few commands thru to the siriproxy but then it started hanging again. I think someone needs to analyze the real packages vs siri proxy packets to see what is or is not being sent to apple.

**Comment by gotnull** [comment]
Last night I was testing my SiriTV plugin and it was working fine. I hadn't tested it for 9 days since it was stuck on FinishSpeech. Maybe Apple are putting in temporary bans? Perhaps the more I use it the quicker it'll stop working?

**Comment by TobyTech** [comment]
We need @plamoni on this.. It is ridiculous. I am 13, set up my own proxy, and got it working completely, but it hangs on finish speech. I have never gotten it on my iPhone, but I do on my iPad.... 0_o

**Comment by TobyTech** [comment]
I am probably wrong, but could it be a dnsmasq issue? Testing now, will report back.

**Comment by TobyTech** [comment]
Never mind. Ignore that ^^

**Comment by TobyTech** [comment]
iPhone over 3G to SiriProxy ----> WORKS
iPhone over local WiFi to SiriProxy ----> WORKS

iPad over 3G/OtherWiFi to SiriProxy ----> FINISHSPEECH!
IPad over local WiFi to SiriProxy ----> FINISHSPEECH!

Conclusion:

IP wasn't banned, UDID Banning??????!?!?!?!?! <--- Going to do more research by spoofing UDID.

Error with non-4S devices with westbaer's fork? <----- NOT TRUE, iPad worked before!

**Comment by RonaldKonijnenburg** [comment]
@TobyTech Does this help:
http://appleworldtoday.com/2012/02/04/fix-siri-proxy-problems/

**Comment by Lunat1k** [comment]
Hmm interesting. That does seem to help. I compared old and new plist copies and there are a few things different between the two. Ace host and Assistant IDs changed. The old copy has a property of "Needs Destruction" = 1; Not sure what that is. But this does bring some interesting questions to the table.

**Comment by RonaldKonijnenburg** [comment]
Needs destruction. I noticed that to!

**Comment by TobyTech** [comment]
I DID notice that as well, and put in it the iPad plist. It didn't help. However, it seems my "temp ban"(or whatever is happening) has finally lifted. Siri is working fine, and I just jinxed (Had to look up how to spell jinxed) myself. I will take a look at that article @sjako thanks!

**Comment by TobyTech** [comment]
Hmm, @sjacko   That does seem like a good plan, I will try that next time. It was so simple, it skipped my mind!

**Comment by TobyTech** [comment]
Sorry @Lunat1k  but I have to ask. Is your name a reference to the watch by Scott Wilson, or just a fun random name?

http://lunatik.com/

**Comment by chickentw** [comment]
How to setup proxychains? I install the tor and proxy chains and edit the proxychains.conf add on some proxies

But every time I use Siri..It still go 127.0.0.1 |D-chain|-<>-127.0.0.1:9050-<><>-17.174.4.4:443-<><>-OK

But I think my IP is block by apple because apple does not send back data and stuck on finish speech

**Comment by RonaldKonijnenburg** [comment]
Delete plist (see above)

**Comment by TobyTech** [comment]
It seems to be working.... for now...;)

**Comment by TobyTech** [comment]
...ANDDDD I jinxed it. It is back and @sjako plist tip didnt work........ Grrrrr.

**Comment by RonaldKonijnenburg** [comment]
Respring after delete

**Comment by gotnull** [comment]
SiriProxy still working for me. I haven't modified anything, I just don't use it every day.

**Comment by TobyTech** [comment]
@sjako I did, and rebooted. Still nothing. Will try again. Still not working. >.<

**Comment by TobyTech** [comment]
@sjako Fixed. That DID solve it. But I noticed something. Here is EXACTLY how I fixed it.
1. Delete assistant.plist
2. Reboot
3. Fire up your 4S and (Using the iFile bluetooth feature) copy THE IPHONES ASSISTANT.PLIST file to the iPad, and replace the existing one.

There you go! That fixed mine. 

BTW. What use is this entire SiriProxy thing? I guarantee the iPad 3 will have Siri... Still, it is nice to boast to my friends who now think I am a professional h@¢ker for getting Siri on my iPad.

My new phrase: "I had Siri on my iPad before all of you! MUAHAHA!"

**Comment by amilham** [comment]
My SiriProxy just started working again, on its own with no changes I made. Hopefully it stays online.

**Comment by plamoni** [comment]
Closing this because it's not really an issue. Feel free to continue discussion of IP bans and whatnot.

**Comment by LewisSmallwood** [comment]
I know this has been closed a while ago, but I thought I might bring it back up as I had the issue today. I also found that one device worked and the other did not. After replacing the .pem file on the other device it worked again. I think that the issue could be something to do with the Config profiles as notmyname4 suggested.

Alternatively, I was having a discussion on another fourm somewhere and I think Apple may be blocking AssistantIds which have been overusing the server during that day. As AssistantIds reset every 24 Hours the problem fixed itself for another person. 

Although, it seems like newly added devices also are more temperamental. (Possibly because of bug fixes in later versions) and because I'm running 6.1.3 on iPad it worked but 4S did not.

I think the 4S issue caused the server to crash after a downgrade from beta software as it stopped the iPad from responding. (I think it may have affected AssistantIds by downgrading from beta software).

^ It's not really an issue, but possibly something which plamoni may experience in the future beta versions (Thought I'd let you know). 'Nuf Said.
- Lewis
